src/api-service/__app__/timer_proxy/__init__.py

```diff
@@ -50,3 +51,10 @@ def main(mytimer: func.TimerRequest) -> None:  # noqa: F841
     for region in regions:
         if all(x.outdated for x in proxies if x.region == region):
             Proxy.get_or_create(region)
+
+    # if there are NSGs with name same as the region that they are allocated
+    # and have no NIC associated with it then delete the NSG
+    for nsg in list_nsgs():
```

Comment:
I think it would be valuable to factor out the logic that decides to delete the region or not as a pure fucnction, then unit test it.